src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1064,15 +1064,24 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
 
       case Token.STRING:
         if (right.isString()) {
-          // Only eval if they are the same type
+          return compareStrings(op, left, right);
+        } else if (right.isNumber()) {
+          // See http://www.ecma-international.org/ecma-262/6.0/#sec-strict-equality-comparison
           switch (op) {
```

Comment:
I'd prefer a switch (and links to the spec) for the cases that can be compared as numbers, rather than the cases that can't.

In fact, can't we merge this case with the following one?